### PR TITLE
Added junit BundleServiceTest - SLIM-522

### PR DESCRIPTION
**Pull request by robinbakkerus**
- In order to support this new Junit test all command executors in 
- BundleService now have an interface with the original name and an
- implementation that ends with Impl.

### CODE REVIEW

**Comment by CLAAdmin** [comment]
[CLA assistant check](http://52.16.237.165:80/OSGP/Protocol-Adapter-DLMS?pullRequest=102) <br/>You should sign our Contributor License Agreement in order to get your pull request merged.<br/>

File: osgp-protocol-adapter-dlms/src/main/java/org/osgp/adapter/protocol/dlms/domain/commands/GetPeriodicMeterReadsGasCommandExecutor.java [hunk 1]
```diff
@@ -119,7 +119,7 @@
          *
          * {8,0-0:1.0.0.255,2,0};                                    position 0
          * {3,1-0:1.8.1.255,2,0};                                    position 1
-         * {3,1-0:1.8.2.255,2,0};                                    position 2
+         * {3,1-0:1.8.2.255,2,0};                                    position 2GetPeriodicMeterReadsGasCommandExecutor
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/main/java/org/osgp/adapter/protocol/dlms/domain/commands/GetPeriodicMeterReadsGasCommandExecutor.java]: doesn't belong here

File: osgp-protocol-adapter-dlms/src/main/java/org/osgp/adapter/protocol/dlms/domain/commands/GetPeriodicMeterReadsGasCommandExecutor.java [hunk 2]
```diff
@@ -153,7 +153,7 @@
          * When no specific capture_objects are selected with selective access:
          *
          * {8,0-0:1.0.0.255,2,0};                                    position 0
-         * {1,0-0:96.10.2.255,2,0}                                   position 1
+         * {1,0-0:96.10.2.255,2,0}                                   position 1GetPeriodicMeterReadsGasCommandExecutor
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/main/java/org/osgp/adapter/protocol/dlms/domain/commands/GetPeriodicMeterReadsGasCommandExecutor.java]: doesn't belong here

File: osgp-protocol-adapter-dlms/src/main/java/org/osgp/adapter/protocol/dlms/domain/commands/SetAlarmNotificationsCommandExecutor.java [hunk 3]
```diff
@@ -216,4 +216,6 @@ public long alarmFilterLongValue(final AlarmNotificationsDto alarmNotifications)
 
         return bitSet.toLongArray()[0];
     }
+
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/main/java/org/osgp/adapter/protocol/dlms/domain/commands/SetAlarmNotificationsCommandExecutor.java]: not necessary

File: osgp-protocol-adapter-dlms/src/main/java/org/osgp/adapter/protocol/dlms/domain/commands/SetPushSetupAlarmBundleCommandExecutor.java [hunk 4]
```diff
@@ -1,57 +1,8 @@
-/**
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/main/java/org/osgp/adapter/protocol/dlms/domain/commands/SetPushSetupAlarmBundleCommandExecutor.java]: header musn't be deleted

File: osgp-protocol-adapter-dlms/src/main/java/org/osgp/adapter/protocol/dlms/domain/commands/GetActualMeterReadsBundleCommandExecutor.java [hunk 5]
```diff
@@ -1,46 +1,8 @@
-/**
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/main/java/org/osgp/adapter/protocol/dlms/domain/commands/GetActualMeterReadsBundleCommandExecutor.java]: header must not be deleted
applicable to all deleted content in the other files.

File: osgp-protocol-adapter-dlms/src/test/java/com/alliander/osgp/dto/valueobjects/smartmetering/ActionDtoBuilder.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
+package com.alliander.osgp.dto.valueobjects.smartmetering;
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/test/java/com/alliander/osgp/dto/valueobjects/smartmetering/ActionDtoBuilder.java]: header needed in file

File: osgp-protocol-adapter-dlms/src/test/java/com/alliander/osgp/dto/valueobjects/smartmetering/ActionDtoBuilder.java [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.joda.time.DateTime;
+
+/**
+ * Helper class to create ActionDto objects.
+ * @author dev
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/test/java/com/alliander/osgp/dto/valueobjects/smartmetering/ActionDtoBuilder.java]: Can be removed. we are all 'dev'

File: osgp-protocol-adapter-dlms/src/test/java/com/alliander/osgp/dto/valueobjects/smartmetering/ActionDtoBuilder.java [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+ *
+ */
+public class ActionDtoBuilder {
+
+    public FindEventsQueryDto makeFindEventsQueryDto() {
+        FindEventsQueryDto result = new FindEventsQueryDto(EventLogCategoryDto.STANDARD_EVENT_LOG, new DateTime(),
+                new DateTime());
+        return result;
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/test/java/com/alliander/osgp/dto/valueobjects/smartmetering/ActionDtoBuilder.java]: return the object immediately. Sonar will complain.
This counts for all the methods here below as well.

File: osgp-protocol-adapter-dlms/src/test/java/com/alliander/osgp/dto/valueobjects/smartmetering/ActionDtoBuilder.java [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+    public ActivityCalendarDataDto makeActivityCalendarDataDto() {
+        final List<SeasonProfileDto> profiles = new ArrayList<>();
+        final ActivityCalendarDto activity = new ActivityCalendarDto("todo", makeCosemDateTimeDto(), profiles);
+        final ActivityCalendarDataDto result = new ActivityCalendarDataDto(activity);
+        return result;
+    }
+    
+    public GMeterInfoDto makeGMeterInfoDto() {
+        final GMeterInfoDto result = new GMeterInfoDto(1, "todo");
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/test/java/com/alliander/osgp/dto/valueobjects/smartmetering/ActionDtoBuilder.java]: Ok, but the following makes more sense:
final GMeterInfoDto result = new GMeterInfoDto(1, "deviceIndentifcation");
or
final GMeterInfoDto result = new GMeterInfoDto(1, "EXXXX001692675614");

File: osgp-protocol-adapter-dlms/src/test/java/com/alliander/osgp/dto/valueobjects/smartmetering/ActionDtoBuilder.java [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+                windowElemenents, 1, 1, 1);
+        final SetPushSetupAlarmRequestDataDto result = new SetPushSetupAlarmRequestDataDto(pushAlarm);
+        return result;
+    }
+
+    public SetPushSetupSmsRequestDataDto mkeSetPushSetupSmsRequestDataDto() {
+        final CosemObisCodeDto cosemCode = new CosemObisCodeDto(1, 1, 1, 1, 1, 1);
+        final List<CosemObjectDefinitionDto> objectDefinitions = new ArrayList<>();
+        final SendDestinationAndMethodDto destinationAndMethod = new SendDestinationAndMethodDto(
+                TransportServiceTypeDto.TCP, "todo", MessageTypeDto.XML_ENCODED_X_DLMS_APDU);
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/test/java/com/alliander/osgp/dto/valueobjects/smartmetering/ActionDtoBuilder.java]: todo is actual a destination

File: osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/application/services/BundleServiceTest.java [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+//        ActualMeterReadsDataDto.class, 
+//        ActualMeterReadsDataGasDto.class, 
+//        SpecialDaysRequestDataDto.class, 
+//        ReadAlarmRegisterDataDto.class, 
+//        GetAdministrativeStatusDataDto.class, 
+//        PeriodicMeterReadsRequestDataDto.class, 
+//        PeriodicMeterReadsGasRequestDataDto.class, 
+//        AdministrativeStatusTypeDataDto.class, 
+//        ActivityCalendarDataDto.class, 
+//        GMeterInfoDto.class,
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/application/services/BundleServiceTest.java]: commented out code can be removed.

File: osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/application/services/BundleServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,231 @@
...
+//        SetPushSetupAlarmRequestDataDto.class, 
+//        SetPushSetupSmsRequestDataDto.class, 
+//        SynchronizeTimeRequestDataDto.class, 
+//        GetConfigurationRequestDataDto.class,
+//        GetFirmwareVersionRequestDataDto.class 
+//    };
+
+    private ActionDtoBuilder builder = new ActionDtoBuilder();
+ 
+    private static Map<Class<?>, AbstractCommandExecutorStub> stubs = new HashMap<Class<?>, AbstractCommandExecutorStub>();
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/application/services/BundleServiceTest.java]: Code 
`Map<Class<?>`
should be
`Map<Class<?` extends ActionDto>

File: osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/domain/commands/stub/AbstractCommandExecutorStub.java [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+import org.osgp.adapter.protocol.dlms.domain.entities.DlmsDevice;
+import org.osgp.adapter.protocol.dlms.exceptions.ProtocolAdapterException;
+
+import com.alliander.osgp.dto.valueobjects.smartmetering.ActionDto;
+import com.alliander.osgp.dto.valueobjects.smartmetering.ActionResponseDto;
+
+public abstract class AbstractCommandExecutorStub {
+
+    private ActionResponseDto actionResponse;
+    private ProtocolAdapterException protocolAdapterException;
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/domain/commands/stub/AbstractCommandExecutorStub.java]: protocolAdapterException is never instantiated.

File: osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/domain/commands/stub/AbstractCommandExecutorStub.java [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+    
+    protected ActionResponseDto doExecute(ClientConnection conn, DlmsDevice device, ActionDto object)
+            throws ProtocolAdapterException {
+
+        if (this.protocolAdapterException != null) {
+            throw protocolAdapterException;
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/domain/commands/stub/AbstractCommandExecutorStub.java]: protocolAdapterException is not instantiated. --> throw null

File: osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/application/services/BundleServiceTest.java [hunk 15]
```diff
@@ -0,0 +1,231 @@
...
+                (AbstractCommandExecutorStub) retrieveConfigurationObjectsBundleCommandExecutor);
+        stubs.put(GetFirmwareVersionRequestDataDto.class,
+                (AbstractCommandExecutorStub) getFirmwareVersionsBundleCommandExecutor);
+    }
+
+    @Test
+    public void testHappyFlow() throws ProtocolAdapterException {
+        BundleMessageDataContainerDto dto = new BundleMessageDataContainerDto(makeActions());
+        List<ActionResponseDto> result = callExecutors(dto);
+        Assert.assertTrue(result != null);
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/application/services/BundleServiceTest.java]: I would add the check of the list size:
`
        final List<ActionDto> actionDtoList = this.makeActions();
        final BundleMessageDataContainerDto dto = new BundleMessageDataContainerDto(actionDtoList);
        final List<ActionResponseDto> result = this.callExecutors(dto);
        Assert.assertTrue(result != null);
        Assert.assertEquals(actionDtoList.size(),  result.size());
`

File: osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/application/services/BundleServiceTest.java [hunk 16]
```diff
@@ -0,0 +1,231 @@
...
+    }
+
+    @Test
+    public void testException() {
+        BundleMessageDataContainerDto dto = new BundleMessageDataContainerDto(makeActions());
+        getStub(FindEventsQueryDto.class).failWith(new ProtocolAdapterException("simulate error"));
+        List<ActionResponseDto> result = callExecutors(dto);
+        Assert.assertTrue(result != null);
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/application/services/BundleServiceTest.java]: also check size. See here above

File: osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/domain/commands/CommandExecutorStub.java [hunk 17]
```diff
@@ -0,0 +1,21 @@
+package org.osgp.adapter.protocol.dlms.domain.commands;
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/domain/commands/CommandExecutorStub.java]: this f

File: osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/domain/commands/CommandExecutorStub.java [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
+import org.openmuc.jdlms.ClientConnection;
+import org.osgp.adapter.protocol.dlms.domain.entities.DlmsDevice;
+import org.osgp.adapter.protocol.dlms.exceptions.ProtocolAdapterException;
+
+import com.alliander.osgp.dto.valueobjects.smartmetering.ActualMeterReadsQueryDto;
+import com.alliander.osgp.dto.valueobjects.smartmetering.MeterReadsDto;
+
+// Dit werkt niet.
+
+public class CommandExecutorStub implements CommandExecutor<ActualMeterReadsQueryDto, MeterReadsDto>  {
```
Comment (hannek24) [inline osgp-protocol-adapter-dlms/src/test/java/org/osgp/adapter/protocol/dlms/domain/commands/CommandExecutorStub.java]: Remove class. This class is used nowhere

**Pull request merged**